wremnants/histselections.py

```diff
@@ -15,39 +15,37 @@
     "ptll_mll" : "nominal",
 }
 
-def fakeHistABCD(h, low_PU=False,
-                 boolMT=True, thresholdMT=40.0, axisNameMT="mt", integrateMT=True):
+def fakeHistABCD(h, thresholdMT=40.0, fakerate_axes=["pt", "eta", "charge"], axis_name_mt="mt", integrateMT=False):
     # boolMT=False generalizes the method when mT is a real axis (i.e. no need for passMT)
     # integrateMT=False keeps the mT axis in the returned histogram (can be used to have fakes vs mT)
-    if low_PU and "mt" in [ax.name for ax in h.axes]:
-        return h[{**failIso, **passMT}]*h[{**passIso, **failMT}].sum().value / h[{**failIso, **failMT}].sum().value
-
-    elif boolMT:
-        return hh.multiplyHists(
-            hh.divideHists(h[{**passIso, **failMT}],
-                           h[{**failIso, **failMT}],
-                           cutoff=1,
-                           createNew=True,
-                           ),
-            #where=h[{"passIso" : False, "passMT" : True}].values(flow=True)>1),
-            h[{"passIso" : False, "passMT" : True}],
-        )
-    else:
+
+    if axis_name_mt in h.axes.name:
         s = hist.tag.Slicer()
-        hFRF = hh.divideHists(h[{**passIso,  axisNameMT : s[0:complex(0,thresholdMT):hist.sum]}], 
-                              h[{**failIso, axisNameMT : s[0:complex(0,thresholdMT):hist.sum]}],
-                              cutoff=1,
-                              createNew=True,
-                              )
+        low = hist.underflow if h.axes[axis_name_mt].traits.underflow else 0
+        failMT = {axis_name_mt : s[low:complex(0,thresholdMT):hist.sum]}
         if integrateMT:
-            return hh.multiplyHists(hFRF,
-                                    h[{**failIso, axisNameMT : s[complex(0,thresholdMT):hist.overflow:hist.sum]}]
-                                    )
+            passMT = {axis_name_mt : s[complex(0,thresholdMT):hist.overflow:hist.sum]}
         else:
-            return hh.multiplyHists(hFRF,
-                                    h[{**failIso, axisNameMT : s[complex(0,thresholdMT):hist.overflow]}],
-                                    )
-        
+            passMT = {axis_name_mt : s[complex(0,thresholdMT):hist.overflow]}
+    else:
+        failMT = common.failMT
+        passMT = common.passMT
+    
+    for a in fakerate_axes:
+        if a not in h.axes.name:
+            logger.warning(f"Fakerate axis '{a}' not found in histogram with axes '{h.axes.name}', try to compute fakerate w/o this axis.")
+            fakerate_axes.remove(a)
+
+    hPassIsoFailMT = h[{**common.passIso, **failMT}]
+    hFailIsoFailMT = h[{**common.failIso, **failMT}]
+
+    if any(a not in fakerate_axes for a in hPassIsoFailMT.axes.name):
+        hPassIsoFailMT = h[{**common.passIso, **failMT}].project(*fakerate_axes)
+        hFailIsoFailMT = h[{**common.failIso, **failMT}].project(*fakerate_axes)
+
+    fakerate = hh.divideHists(hPassIsoFailMT, hFailIsoFailMT, cutoff=1, createNew=True)
```

Comment:
Minor, but could we call this "fakerateFactor" or anything different from "fakerate"? Just to avoid confusion in the future if a new user checks the code, since the ratio that is being computed is not the fakerate (although it is still a monotonic function of it). The actual fakerate would have been defined as pass/(pass+fail)